ui/components/app/confirmations/alerts/multiple-alert-modal/multiple-alert-modal.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import {
+  Box,
+  ButtonIcon,
+  ButtonIconSize,
+  IconName,
+  Text,
+} from '../../../../component-library';
+import {
+  Display,
+  TextColor,
+  TextVariant,
+} from '../../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import useAlerts from '../../../../../hooks/useAlerts';
+import { AlertModal } from '../alert-modal';
+import { Alert } from '../../../../../ducks/confirm-alerts/confirm-alerts';
+
+export type MultipleAlertModalProps = {
+  /** The key of the initial alert to display. */
+  alertKey: string;
+  /** The function to be executed when the modal needs to be closed. */
+  onClose: () => void;
+  /** The function to be executed when the button in the alert modal is clicked. */
+  onFinalAcknowledgeClick: () => void;
+  /** The unique identifier of the entity that owns the alert. */
+  ownerId: string;
+};
+
+function PageNavigation({
+  alerts,
+  onBackButtonClick,
+  onNextButtonClick,
+  selectedIndex,
+}: {
+  alerts: Alert[];
+  onBackButtonClick: () => void;
+  onNextButtonClick: () => void;
+  selectedIndex: number;
+}) {
+  const t = useI18nContext();
+  const showPreviousButton = selectedIndex + 1 > 1;
+  const showNextButton = selectedIndex + 1 < alerts.length;
+
+  function PreviousButton() {
```

Comment:
Minor, but can we define these outside of the `PageNavigation` component to decouple them a little?

We could pass the `selectedIndex` to each so they can do their own checks if they need to render?